test/dataframe.jl

```diff
@@ -58,6 +58,7 @@ module TestDataFrame
 
     z = vcat(v, x)
 
+    # TODO: change test to throw an error after deprecation
```

Comment:
Without more details we may well miss it when removing the deprecation. Same below. Would be good to add a reference to the PR which introduced the deprecation, and a comment in deprecated.jl saying to look for them. Even better, prepare the code and keep it commented out while it's fresh in your mind, so that we know what to do later.

Reply:
Fixed - now deprecations are thrown in both cases so when we are past deprecation period they will not be missed as tests will throw an error.

Additionally in each case I have added expected error thrown in source (to be uncommented when removing deprecation) and accompanying test for this error.

Finally I removed two methods from *deprecated.jl* as they are covered in other places with a proper deprecation.